mlflow/models/evaluation/base.py

```diff
@@ -0,0 +1,530 @@
+import time
+import numpy as np
+import pandas as pd
+import json
+import os
+from mlflow.exceptions import MlflowException
+from mlflow.utils.file_utils import TempDir
+from mlflow.entities import Metric, RunTag
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils import _get_fully_qualified_class_name
+from mlflow.utils.class_utils import _get_class_from_string
+import logging
+
+
+_logger = logging.getLogger(__name__)
+
+
+class EvaluationMetrics(dict):
+    """
+    Represent a dict of metrics.
+    """
+
+    pass
+
+
+class EvaluationArtifact:
+    """
+    Represent a artifact. Contains artifact uri and content.
+    """
+
+    def __init__(self, uri, content=None):
+        self._uri = uri
+        self._content = content
+
+    def _load_content_from_file(self, local_artifact_path):
+        raise NotImplementedError()
+
+    def load(self, local_artifact_path=None):
+        """
+        If `local_artifact_path` is None, download artifact from the artifact uri and load it.
+        otherwise load artifact content from specified path.
+        """
+        if local_artifact_path is None:
+            return self._load_content_from_file(local_artifact_path)
+        else:
+            with TempDir() as temp_dir:
+                local_artifact_file = temp_dir.path("local_artifact")
+                _download_artifact_from_uri(self._uri, local_artifact_file)
+                self._load_content_from_file(local_artifact_file)
+
+    def save(self, output_artifact_path):
+        """Save artifact content into specified path."""
+        raise NotImplementedError()
+
+    @property
+    def content(self):
+        """
+        The content of the artifact (representation varies)
+        """
+        if self._content is None:
+            self.load()
+        return self._content
+
+    @property
+    def uri(self) -> str:
+        """
+        The URI of the artifact
+        """
+        return self._uri
+
+
+class EvaluationResult:
+    """
+    Represent an return value of `mlflow.evaluate()` API. Contains metrics dict and
+    artifact dict.
+    """
+
+    def __init__(self, metrics, artifacts):
+        self._metrics = metrics
+        self._artifacts = artifacts
+
+    @classmethod
+    def load(cls, path):
+        """Load the evaluation results from the specified local filesystem path"""
+        with open(os.path.join(path, "metrics.json"), "r") as fp:
+            metrics = EvaluationMetrics(json.load(fp))
+
+        with open(os.path.join(path, "artifacts_metadata.json"), "r") as fp:
+            artifacts_metadata = json.load(fp)
+
+        artifacts = {}
+
+        artifacts_dir = os.path.join(path, "artifacts")
+
+        for artifact_name, meta in artifacts_metadata:
+            location = meta["location"]
+            ArtifactCls = _get_class_from_string(meta["class_name"])
+            content = ArtifactCls.load_content_from_file(os.path.join(artifacts_dir, artifact_name))
```

Comment:
Can we call `ArtifactCls.load()` and pass in `local_artifact_path`, rather than calling the internal method?